index.js

```diff
@@ -55,8 +55,8 @@ function _writeFile (filename, data, options, callback) {
       options.mode && [fs, fs.chmod, tmpfile, options.mode],
       [fs, fs.rename, tmpfile, filename]
     ], function (err) {
-      err ? fs.unlink(tmpfile, function () { callback(err) })
-        : callback()
+      err ? fs.unlink(tmpfile, function () { callback(err, tmpfile) })
+        : callback(null, tmpfile)
```

Comment:
These aren't called until the whole process completes (and the tmpfile no longer exists), so I don't think this will do what you want it to do…

Reply:
Reading the PR description, I think this does work for us (AVA) because we introduce a slight delay between detecting file changes and acting on them. Knowing the`tmpfile` allows us to disregard those changes.